src/test/java/org/checkstyle/autofix/recipe/AbstractRecipeTestSupport.java

```diff
@@ -56,75 +62,88 @@ protected String getPackageLocation() {
         return "org/checkstyle/autofix/recipe/" + getSubpackage();
     }
 
-    protected void verify(String testCaseName) throws Exception {
+    protected static Stream<Arguments> reportParsers() {
+        return Stream.of(
+                Arguments.of(Named.of("XML", new XmlReportParser())),
+                Arguments.of(Named.of("SARIF", new SarifReportParser()))
+        );
+    }
 
+    protected void verify(ReportParser parser, String testCaseName) throws Exception {
         final String inputPath = getInputFilePath(testCaseName);
         final String outputPath = getOutputFilePath(testCaseName);
         final Configuration config = getCheckConfigurations(inputPath);
         verifyOutputFile(outputPath, config);
 
-        final Path violations = runCheckstyle(inputPath, config);
-        final Path configPath = createConfigFile(config);
-
+        final Path reportPath = runCheckstyle(parser, inputPath, config);
         verifyWithInlineConfigParser(getPath(inputPath),
-                convertToExpectedMessages(CheckstyleReportParser.parse(violations)));
-
-        final String beforeCode = readFile(getPath(inputPath));
-        final String expectedAfterCode = readFile(getPath(outputPath));
-
-        try {
-            final Recipe mainRecipe = new CheckstyleAutoFix(violations.toString(),
-                    configPath.toString());
-            testRecipe(beforeCode, expectedAfterCode,
-                    getPath(inputPath), new InputClassRenamer(),
-                    mainRecipe, new RemoveViolationComments());
-        }
-        finally {
-            Files.deleteIfExists(configPath);
-            Files.deleteIfExists(violations);
-        }
+                convertToExpectedMessages(parser.parse(reportPath)));
+        verify(config, reportPath, inputPath, outputPath);
     }
 
-    protected void verify(Configuration config, String testCaseName) throws Exception {
+    protected void verify(ReportParser parser, Configuration config, String testCaseName)
+            throws Exception {
         final String inputPath = getInputFilePath(testCaseName);
         final String outputPath = getOutputFilePath(testCaseName);
-
         verifyOutputFile(outputPath, config);
 
-        final Path violationReportPath = runCheckstyle(inputPath, config);
-
-        final String beforeCode = readFile(getPath(inputPath));
-        final String expectedAfterCode = readFile(getPath(outputPath));
+        final Path reportPath = runCheckstyle(parser, inputPath, config);
```

Comment:
Using parser here sounds weird.
In code we don't not use instance at all, and we just need xml/sarif  flag .
I am not sure why we need instances of parser, but to run checkstyle we should just give enum like type  to Create xml of sarif